From: Lokesh Gidra <lokeshgidra@google.com>
Subject: userfaultfd-opportunistic-tlb-flush-batching-for-present-pages-in-move-v6
Date: Sat, 16 Aug 2025 12:11:23 -0700

make calculation of largest extent that can be batched unconditional on
length, per Barry

Link: https://lkml.kernel.org/r/20250816191123.3601561-1-lokeshgidra@google.com
Signed-off-by: Lokesh Gidra <lokeshgidra@google.com>
Acked-by: Peter Xu <peterx@redhat.com>
Reviewed-by: Barry Song <baohua@kernel.org>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Kalesh Singh <kaleshsingh@google.com>
Cc: Barry Song <v-songbaohua@oppo.com>
Cc: David Hildenbrand <david@redhat.com>
Cc: Peter Xu <peterx@redhat.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/userfaultfd.c |    7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

--- a/mm/userfaultfd.c~userfaultfd-opportunistic-tlb-flush-batching-for-present-pages-in-move-v6
+++ a/mm/userfaultfd.c
@@ -1081,11 +1081,8 @@ static long move_present_ptes(struct mm_
 	unsigned long src_start = src_addr;
 	unsigned long src_end;
 
-	if (len > PAGE_SIZE) {
-		len = pmd_addr_end(dst_addr, dst_addr + len) - dst_addr;
-		src_end = pmd_addr_end(src_addr, src_addr + len);
-	} else
-		src_end = src_addr + len;
+	len = pmd_addr_end(dst_addr, dst_addr + len) - dst_addr;
+	src_end = pmd_addr_end(src_addr, src_addr + len);
 	flush_cache_range(src_vma, src_addr, src_end);
 	double_pt_lock(dst_ptl, src_ptl);
 
_
